sdks/java/core/src/main/java/org/apache/beam/sdk/util/NullCredential.java

```diff
@@ -0,0 +1,57 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.apache.beam.sdk.util;
+
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpResponseInterceptor;
+import java.io.IOException;
+
+/**
+ * A {@link HttpRequestInitializer} for requests that don't have credentials.
+ *
+ * <p>When the access is denied, it throws {@link IOException} with a detailed error message.
+ */
+public class NullCredential implements HttpRequestInitializer {
```

Comment:
I prefer to keep NullCredential.

SpecificExceptionAfterNonauthorizedResponse is describing the implementation. If we need to do more for a "NullCredential", we will have to change the name. And, NullCredential follows the current pattern, and reads better in Transport as following:
if (credential == null) {
  return new ChainingHttpRequestInitializer(new NullCredential(), httpRequestInitializer);
} else {
  return new ChainingHttpRequestInitializer(credential, httpRequestInitializer);
}